src/features/wallet/TwoKeyContractModal.tsx

```diff
@@ -24,8 +38,43 @@ const _TwoKeyContractModal: ForwardRefRenderFunction<
   { isOpen, close, hotAddress, votingAddress, isConnected, network, signer },
   externalRef
 ) => {
+  const [showForm, setShowForm] = useState(false);
+  const [value, setValue] = useState("");
+  const [error, setError] = useState("");
+  const [success, setSuccess] = useState(false);
+  const closeForm = useCallback(() => {
+    if (showForm) {
+      setShowForm(false);
+      setValue("");
+    } else {
+      setSuccess(false);
+      setShowForm(true);
+    }
+  }, [showForm]);
+
+  const submitForm = useCallback(() => {
+    if (value.substr(0, 2).toLowerCase() !== "0x")
+      return setError("Address must start with 0x");
+    if (value.length !== 42) return setError("Address must be 42 characters");
+    if (value && network && signer) {
+      return createDesignatedVotingContract(value, signer, network).then(
```

Comment:
would highly recommend catching an error and doing something with it, even if its just to log

Reply:
Good call.